runtime/vm/resolvefield.cpp

```diff
@@ -1032,6 +1032,13 @@ fieldOffsetsNextDo(J9ROMFieldOffsetWalkState *state)
 		/* Note: hiddenInstanceFieldWalkIndex is the index of the last hidden instance field that was returned. */
 		J9HiddenInstanceField *hiddenField = state->hiddenInstanceFields[--state->hiddenInstanceFieldWalkIndex];
 
+		if (J9_ARE_ALL_BITS_SET(state->walkFlags, J9VM_FIELD_OFFSET_WALK_ONLY_OBJECT_SLOTS)) {
+			if (J9_ARE_NO_BITS_SET(hiddenField->shape->modifiers, J9FieldFlagObject)) {
+				/* if its not an O-Slot skip to the next field */
+				return fieldOffsetsNextDo(state);
+			}
```

Comment:
I think it would be better to locally iterate through the `state->hiddenInstanceFields[]` rather than calling the method again.

Something vaguely like:
```
while (0 != state->hiddenInstanceFieldWalkIndex) {
	if (J9_ARE_NO_BITS_SET(hiddenField->shape->modifiers, J9FieldFlagObject)) {
		/* if its not an O-Slot skip to the next field */
		hiddenField = state->hiddenInstanceFields[--state->hiddenInstanceFieldWalkIndex];
	} else {
		/* Found an OSlot */
		state->result.field = hiddenField->shape;
		..... other existing code ...
		break;
	}
}

Reply:
At a minimum, the existing pattern will unbalance the tracepoints

Reply:
Does this change need to be mirrored in DDR? The createramclass one clearly does not.

Reply:
I think DDR should remain consistent, Ill make that change as well